features/FEATURE_BLE/targets/TARGET_CORDIO/source/CordioBLE.cpp

```diff
@@ -388,20 +393,22 @@ void BLE::callDispatcher()
     // process the external event queue
     _event_queue.process();
 
-    // follow by stack events
-    static uint32_t lastTimeUs = us_ticker_read();
-    uint32_t currTimeUs, deltaTimeMs;
+    _last_update_us += (uint64_t)_timer.read_high_resolution_us();
+    _timer.reset();
 
-    // Update the current cordio time
-    currTimeUs = us_ticker_read();
-    deltaTimeMs = (currTimeUs - lastTimeUs) / 1000;
-    if (deltaTimeMs > 0) {
-        WsfTimerUpdate(deltaTimeMs / WSF_MS_PER_TICK);
-        lastTimeUs += deltaTimeMs * 1000;
+    uint64_t last_update_ms   = (_last_update_us / 1000);
+    wsfTimerTicks_t wsf_ticks = (last_update_ms / WSF_MS_PER_TICK);
+
+    if (wsf_ticks > 0) {
+        WsfTimerUpdate(wsf_ticks);
+
+        _last_update_us -= (last_update_ms * 1000);
```

Comment:
```suggestion
        _last_update_us -= (wsf_ticks * WSF_MS_PER_TICK * 1000);
```
otherwise you're losing milliseconds that aren't passed in as ticks